source/OctopusTools/Importers/CheckedReferences.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using Octopus.Client.Model;
+
+namespace OctopusTools.Importers
+{
+    public class CheckedReferences<T> where T:Resource
+    {
+        public IDictionary<string, T> FoundDependencies { get; set; }
+        public ICollection<string> MissingDependencyNames { get; set; }
+
+        public List<string> FoundDependencyIds
+        {
+            get { return FoundDependencies.Keys.ToList(); }
+        }
+
+        public IEnumerable<string> MissingDependencyErrors
+        {
+            get { return MissingDependencyNames.Select(d => string.Format("Could not find {0} '{1}'", typeof (T).Name.Replace("Resource",""), d)); }
```

Comment:
If this is generic type <T> where T is always the same, can we resolve its name once rather than in select iteration?